File: app/src/main/java/androidacademy/minsk/lecture3lists/ActorListActivity.java
package androidacademy.minsk.lecture3lists;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import androidacademy.minsk.lecture3lists.data.Actor;
import androidacademy.minsk.lecture3lists.data.DataUtil;
import androidx.appcompat.app.AppCompatActivity;

public class ActorListActivity extends AppCompatActivity {

    private ActorListAdapter adapter;

    private final AdapterView.OnItemClickListener clickListener = (adapterView, view, position, id) -> {
        Actor actor = adapter.getItem(position);
        String clickMessage = getString(R.string.actor_click_action, actor.getName());

        Toast.makeText(this, clickMessage, Toast.LENGTH_SHORT).show();
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_actor_list);

        ListView list = findViewById(R.id.actorList);
        adapter = new ActorListAdapter(this, DataUtil.generateActors());
        list.setAdapter(adapter);
        list.setOnItemClickListener(clickListener);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.actor_list_menu, menu);

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.switchList:
                startActivity(new Intent(this, ActorRecyclerActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/androidacademy/minsk/lecture3lists/ActorListActivity.kt
package androidacademy.minsk.lecture3lists

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListView
import android.widget.Toast
import androidacademy.minsk.lecture3lists.data.DataUtil.generateActors
import androidx.appcompat.app.AppCompatActivity

class ActorListActivity : AppCompatActivity() {

    private lateinit var adapter: ActorListAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_actor_list)

        val list = findViewById<ListView>(R.id.actorList)
        adapter = ActorListAdapter(this, generateActors())
        list.adapter = adapter
        list.onItemClickListener = OnItemClickListener { _, _, position, _ ->
            val actor = adapter.getItem(position)
            val clickMessage = getString(R.string.actor_click_action, actor.name)
            Toast.makeText(this, clickMessage, Toast.LENGTH_SHORT).show()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.actor_list_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.switchList -> {
                startActivity(Intent(this, ActorRecyclerActivity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}